src/Database/SchemaCache.php

```diff
@@ -89,11 +90,11 @@ public function clear(?string $name = null): array
     /**
      * Helper method to get the schema collection.
      *
-     * @param \Cake\Database\Connection $connection Connection object
+     * @param \Cake\Datasource\ConnectionInterface $connection Connection object
      * @return \Cake\Database\Schema\Collection|\Cake\Database\Schema\CachedCollection
      * @throws \RuntimeException If given connection object is not compatible with schema caching
      */
-    public function getSchema(Connection $connection)
+    public function getSchema(ConnectionInterface $connection)
```

Comment:
The method uses `$connection->cacheMetadata()` and `cacheMetadata()` does not exist in `ConnectionInterface`. So the typhint should be reverted to concrete class (and changed in constructor too).

Adding `cacheMetadata()` to the interface isn't an option as for e.g. elastic search plugin's connection class does not have this method.

Reply:
Hm, that is less than ideal.
As this means the whole class needs it, also means we get IDE errors on `ConnectionManager::get('test')` return value and usage between the cache and non cache versions using either concrete or interface contracts.
I made a PR to this one, but I sure dont like it.
Better to somehow add cacheMetadata() and throw not implemented exception or alike IMO.